src/xclim/indices/_hydrology.py

```diff
@@ -647,3 +648,43 @@ def low_flow_frequency(q: xarray.DataArray, threshold_factor: float = 0.2, freq:
     threshold = threshold_factor * mean_flow
     out = threshold_count(q, "<", threshold, freq=freq)
     return to_agg_units(out, q, "count")
+
+
+@declare_units(pr="[precipitation]")
+def antecedent_precipitation_index(pr: xarray.DataArray, window: int = 7, p_exp: float = 0.935) -> xarray.DataArray:
+    """
+    Antecedent Precipitation Index.
+
+    Calculate the running weighted sum of daily precipitation values given a window and weighting exponent.
+    This index serves as an indicator for soil moisture.
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+        Daily precipitation data.
+    window : int, optional
+        Window for the days of precipitation data to be weighted and summed, default is 7.
+    p_exp : float, optional
+        Weighting exponent, default is 0.935.
+
+    Returns
+    -------
+    xarray.DataArray
+        Antecedent Precipitation Index.
+
+    References
+    ----------
+    :cite:cts:`schroter2015,li2021`
+    """
+    # TODO: ensure precip is mm
+    # construct a xr.DataArray containing the weights
+    weights = xarray.DataArray(
+        list(reversed([p_exp ** (idx - 1) for idx in range(1, window + 1)])),
+        dims="window_dim",
+    )
+    # apply the weights to input data
+    out = pr.rolling(time=window).construct("window_dim").dot(weights)
+    out.attrs["window"] = window
+    out.attrs["p_exp"] = p_exp
+    out.attrs["units"] = "mm day-1"
```

Comment:
```suggestion
    out.attrs["units"] = "mm"
```
Following my suggestions above and the articles, I think the API should have units of precipitation totals, no ? That's also what you put for the indicator anyway.

Reply:
Yes, that's correct. I wasn't sure if they temporal frequency needs to specified in the unit or not.